test/comparisons.jl

```diff
@@ -247,20 +263,30 @@ end
 
             # Using a vector of intervals as sets
             @test union([earlier, later]) == [expected_superset]
+            @test union(IntervalSet([earlier, later])) == IntervalSet([expected_superset])
 
             # TODO: These functions should be compatible with unbounded intervals
             if isbounded(earlier) && isbounded(later)
-                @test union([earlier], [later]) == [expected_superset]
+                @test union([earlier], [later]) == expected_xor != [expected_superset]
```

Comment:
For cases like this where something was previously `==` I've included both the `!=` case and what's actually returned. I'm not sure if we should keep these tests in since this is behaviour that was reverted?